### PR TITLE
added min ADA calculation for more precise max value

### PR DESCRIPTION
**Pull request by alessandrokonrad**
I had a few cases, where the max value size was exceeded by a few bytes, because the add_change_if_needed function doesn't take ADA into consideration when it splits the utxos, so I added the minADA to make the calculation more precise.

### CODE REVIEW

File: rust/src/tx_builder.rs [hunk 1]
```diff
@@ -878,7 +882,7 @@ impl TransactionBuilder {
                                 let asset_name = asset_names.get(n);
                                 let value = assets.get(&asset_name).unwrap();
 
-                                if will_adding_asset_make_output_overflow(&output, &rebuilt_assets, (policy.clone(), asset_name.clone(), value), max_value_size) {
+                                if will_adding_asset_make_output_overflow(&output, &rebuilt_assets, (policy.clone(), asset_name.clone(), value), max_value_size, coins_per_utxo_word) {
```
Comment (rooooooooob) [inline rust/src/tx_builder.rs:885]: Note: #304 updates this so you might need to fix that before we can merge this depending on which one gets merged first

**Review by rooooooooob — APPROVED** [review]
LGTM, nice catch

**Review by vsubhuman — APPROVED** [review]
Thank you, @alessandrokonrad !

**Review by vsubhuman — COMMENTED** [review]
/check

**Pull request merged**